vmm/src/api/mod.rs

```diff
@@ -128,6 +128,7 @@ pub struct VmResizeData {
 #[derive(Clone, Deserialize, Serialize)]
 pub struct VmAddDeviceData {
```

Comment:
Remove this and replace it with a `DeviceConfig`. It already has all these fields and the iommu one which is currently missing.

Reply:
I was planning to do that in a follow up PR as I'm going to add the vIOMMU support. Do you want me to do it here or do you think this can wait the follow up?

Reply:
I think we should do it now. I didn't realise you weren't using the same struct as that's the struct i'm using in the ch-remote API call.

Reply:
(Of course it would work as the fields are the same name.)

Reply:
Ok let me do that right now :)